pkg/subctl/cmd/gather/connectivity.go

```diff
@@ -0,0 +1,120 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gather
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/submariner-io/submariner-operator/pkg/discovery/network"
+)
+
+const (
+	gatewayPods = "app=submariner-gateway"
+	routeagentPods = "app=submariner-routeagent"
+)
+
+func GetGatewayPodsLogs(clientSet kubernetes.Interface, clustername string) error {
+	pods, err := network.FindPods(clientSet, v1.ListOptions{LabelSelector: gatewayPods})
+
+	if err != nil || pods == nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		err := getPodLogs(clientSet, pod.Name, clustername)
+		if err != nil {
+			return errors.WithMessagef(err, "error getting logs for pod %q", pod.Name)
+		}
+	}
+	return nil
+}
+
+func GetRouteagentPodsLogs(clientSet kubernetes.Interface, clustername string) error {
+	pods, err := network.FindPods(clientSet, v1.ListOptions{LabelSelector: routeagentPods})
+
+	if err != nil || pods == nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		err := getPodLogs(clientSet, pod.Name, clustername)
+		if err != nil {
+			return errors.WithMessagef(err, "error getting logs for pod %q", pod.Name)
+		}
+	}
+	return nil
+}
+
+func getPodLogs(clientset kubernetes.Interface, podname string, clustername string) error {
+	logrequest := clientset.CoreV1().Pods("submariner-operator").GetLogs(podname, &corev1.PodLogOptions{})
```

Comment:
Let's avoid hard-coding the namespace. We can get it from the `Pod` instance.

```suggestion
func getPodLogs(clientset kubernetes.Interface, pod *corev1.Pod, clustername string) error {
	logrequest := clientset.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{})
```

Reply:
all submariner related pods are in submariner-operator ns. There are multiple instances in the code where the ns is hard-coded as a constant. I will do the same.

Reply:
There probably are but we shouldn't perpetuate that especially since we have the proper namespace in hand and don't need to hard-code. I believe we have a future use case to customize it.